File: Palindrome Number.java

class Solution {
    public boolean isPalindrome(int x) 
    {
        int n = x;
      
        if(x<0) // For negative numbers it will return false
        {
            return false;
        }
        int p = 0;
        while(x>0)
        {
            p = p * 10 + (x%10);
            x /= 10;
            
        }
        return p == n;
        
        
        
        
    }
}

File: 9-palindrome-number/palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """

        if x < 0:
            return False
        if 0 <= x <= 9 :
            return True

        original = x
        reversed = 0
        
        while x > 0:
            digit = x % 10
            reversed = reversed * 10 + digit
            x = x // 10

        return original == reversed

        